presto-main/src/main/java/com/facebook/presto/util/JsonUtil.java

```diff
@@ -201,7 +210,9 @@ private static boolean isValidJsonObjectKeyType(Type type)
                 baseType.equals(StandardTypes.REAL) ||
                 baseType.equals(StandardTypes.DOUBLE) ||
                 baseType.equals(StandardTypes.DECIMAL) ||
-                baseType.equals(StandardTypes.VARCHAR);
+                baseType.equals(StandardTypes.VARCHAR) ||
+                type instanceof LongEnumType ||
```

Comment:
I think we should make the `TypeSignature` of enum types to include the information that they are enums.